### PR TITLE
As SM I wan to use Gin theme

### PR DESCRIPTION
**Pull request by agami4**
## Problem
As SM I wan to use Gin theme

## Solution
Should be possible to use Gin theme like front-end theme

## Issue tracker
https://www.drupal.org/project/social/issues/3142830

## How to test
- [ ] 

## Screenshots
Screenshot at May 27 17-49-24
Screenshot at May 27 17-50-03

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -102,6 +102,9 @@
             "drupal/votingapi" : {
                 "votingapi_views_data_alter uses incorrect empty check for missing entity error handling": "https://www.drupal.org/files/issues/2020-01-31/votingapi-3110353-2.patch",
                 "#3048085 - fix time() callback filling up the logs": "https://www.drupal.org/files/issues/2019-05-02/votingapi-timecallback-3048085-1-4.patch"
+            },
+            "drupal/gin": {
+                "3150481/3150243 - Update layout for the action block on the '/admin/content' page and fix position for the checkbox element": "https://www.drupal.org/files/issues/2020-06-10/3150481-3150243-layout-action-block-and-checkbox.patch"
             }
```
Comment (T2L) [inline composer.json:108]: @agami4 "new VBO patch"

File: composer.json [hunk 2]
```diff
@@ -130,6 +134,8 @@
         "drupal/field_group": "3.0",
         "drupal/file_mdm": "1.1",
         "drupal/flag": "4.0-alpha3",
+        "drupal/gin": "dev-3.x",
```
Comment (T2L) [inline composer.json:137]:
```suggestion
        "drupal/gin": "dev-3.x#451ca6577145323bd7eafcbf9b6175eab1ebeae7",
```

File: drupal-org.make [hunk 3]
```diff
@@ -57,6 +59,10 @@ projects[group][patch][] = "https://www.drupal.org/files/issues/2018-11-01/30108
 projects[group][patch][] = "https://www.drupal.org/files/issues/2019-12-04/group-3098675-2.patch"
 projects[group][patch][] = "https://www.drupal.org/files/issues/2019-07-04/group-request_membership_feature_grequest_module-2752603-110.patch"
 projects[group][patch][] = "https://www.drupal.org/files/issues/2020-03-26/ginvite_module_port-2801603-112.patch"
+projects[gin][type] = theme
+projects[gin][version] = dev-3.x
```
Comment (T2L) [inline drupal-org.make:63]:
```suggestion
projects[gin][branch] = 8.x-3.x
projects[gin][revision] = 451ca6577145323bd7eafcbf9b6175eab1ebeae7
```

File: drupal-org.make [hunk 4]
```diff
@@ -127,3 +133,4 @@ projects[votingapi][patch][] = "https://www.drupal.org/files/issues/2019-05-02/v
 projects[bootstrap][type] = theme
 projects[bootstrap][version] = 3.20
 projects[bootstrap][patch][] = "https://www.drupal.org/files/issues/2018-12-19/dropdown-without-default-button-3021413-2.patch"
+projects[gin][patch][] = "https://www.drupal.org/files/issues/2020-06-10/3150481-3150243-layout-action-block-and-checkbox.patch"
```
Comment (T2L) [inline drupal-org.make:136]:
```suggestion
projects[gin][patch][] = "new VBO patch"
```

File: composer.json (hunk 1 above)
Comment (agami4) [reply composer.json]: done

File: composer.json (hunk 2 above)
Comment (agami4) [reply composer.json:137]: done

File: drupal-org.make (hunk 3 above)
Comment (agami4) [reply drupal-org.make]: done

File: drupal-org.make (hunk 4 above)
Comment (agami4) [reply drupal-org.make]: done

Comment (agami4) [reply drupal-org.make]: done

File: modules/social_features/social_core/social_core.module [hunk 5]
```diff
@@ -46,15 +46,20 @@ function social_core_modules_installed($modules) {
...
   // for the optional view mode by enabling social_flexible_group_featured.
-  if ((in_array('social_group_flexible_group', $modules, TRUE) && \Drupal::moduleHandler()->moduleExists('social_landing_page')) ||
-    (in_array('social_landing_page', $modules, TRUE) && \Drupal::moduleHandler()->moduleExists('social_group_flexible_group'))) {
-    \Drupal::service('module_installer')->install(['social_flexible_group_featured']);
+  if ((in_array('social_group_flexible_group', $modules, TRUE) && \Drupal::moduleHandler()
+    ->moduleExists('social_landing_page')) ||
+    (in_array('social_landing_page', $modules, TRUE) && \Drupal::moduleHandler()
+      ->moduleExists('social_group_flexible_group'))) {
+    \Drupal::service('module_installer')
+      ->install(['social_flexible_group_featured']);
```
Comment (T2L) [inline modules/social_features/social_core/social_core.module:54]: This change is not required

File: modules/social_features/social_core/social_core.module [hunk 6]
```diff
@@ -46,15 +46,20 @@ function social_core_modules_installed($modules) {
...
   }
   // If either social_book or social_landing_page gets enabled and
   // the other is already enabled, we should import the the configuration
   // for the optional view mode by enabling social_book_featured.
-  if ((in_array('social_book', $modules, TRUE) && \Drupal::moduleHandler()->moduleExists('social_landing_page')) ||
-    (in_array('social_landing_page', $modules, TRUE) && \Drupal::moduleHandler()->moduleExists('social_book'))) {
+  if ((in_array('social_book', $modules, TRUE) && \Drupal::moduleHandler()
+    ->moduleExists('social_landing_page')) ||
+    (in_array('social_landing_page', $modules, TRUE) && \Drupal::moduleHandler()
+      ->moduleExists('social_book'))) {
```
Comment (T2L) [inline modules/social_features/social_core/social_core.module:62]: This change is not required

File: tests/behat/features/capabilities/gdpr/create-data-policy.feature [hunk 7]
```diff
@@ -24,7 +24,6 @@ Feature: Create data policy and view new policy
     And I should see the link "Revisions" in the "Tabs" region
     And I should see the text "Revision"
     And I should see the text "Operations"
-    And I should see the text "List is empty."
```
Comment (T2L) [inline tests/behat/features/capabilities/gdpr/create-data-policy.feature:27]: Why has this line been removed?

File: themes/socialbase/assets/css/admin-toolbar.css [hunk 8]
```diff
@@ -1,35 +1,41 @@
 .toolbar-vertical #block-filter:target {
-  transform: translateY(39px) translateX(0);
+  -webkit-transform: translateY(39px) translateX(0);
```
Comment (T2L) [inline themes/socialbase/assets/css/admin-toolbar.css:2]: I don't think `-webkit-` prefixed version is required anymore.

File: themes/socialbase/assets/css/admin-toolbar.css [hunk 9]
```diff
@@ -1,35 +1,41 @@
...
+          transform: translateY(39px) translateX(0);
 }
 
 .toolbar-vertical .off-canvas-right {
-  transform: translateY(39px) translateX(280px);
+  -webkit-transform: translateY(39px) translateX(280px);
```
Comment (T2L) [inline themes/socialbase/assets/css/admin-toolbar.css:7]: I don't think `-webkit-` prefixed version is required anymore.

Comment (agami4) [reply themes/socialbase/assets/css/admin-toolbar.css:7]: This issue is not relevant

File: themes/socialbase/assets/css/admin-toolbar.css (hunk 8 above)
Comment (agami4) [reply themes/socialbase/assets/css/admin-toolbar.css:2]: This issue is not relevant

File: tests/behat/features/capabilities/gdpr/create-data-policy.feature (hunk 7 above)
Comment (agami4) [reply tests/behat/features/capabilities/gdpr/create-data-policy.feature]: done

File: modules/social_features/social_core/social_core.module (hunk 6 above)
Comment (agami4) [reply modules/social_features/social_core/social_core.module:62]: This issue is not relevant

File: modules/social_features/social_core/social_core.module (hunk 5 above)
Comment (agami4) [reply modules/social_features/social_core/social_core.module:54]: This issue is not relevant

**Comment by agami4** [comment]
@ribel, @T2L, tests passed successfully

File: modules/social_features/social_core/social_core.install [hunk 10]
```diff
@@ -1229,3 +1229,17 @@ function social_core_update_8811() {
...
+ */
+function social_core_update_8812(&$sandbox) {
+  // Install Gin Admin Theme.
+  \Drupal::service('theme_installer')->install(['gin']);
+  // Set Gin Admin Theme as admin theme.
+  \Drupal::configFactory()->getEditable('system.theme')
+    ->set('admin', 'gin')
+    ->save();
+  // Install Gin Toolbar and Coffee modules.
+  \Drupal::service('module_installer')->install(['gin_toolbar', 'coffee']);
```
Comment (bramtenhove) [inline modules/social_features/social_core/social_core.install:1244]: Why are we adding the `coffee` module? Won't this make it much more difficult more most users?

Comment (agami4) [reply modules/social_features/social_core/social_core.install:1244]: It was requested from Taco)

**Comment by ronaldtebrake** [comment]
We will remove the coffee module anyway, due to conflicts that can occur with our SaaS modules / Extensions.

**Comment by ronaldtebrake** [comment]
Thanks for all the amazing work, we need to address the following:

1. We didn't lock the hash for the GIN version in our _composer.json_ which means when I installed it, it broke. You can see the tests now also failing, since I retriggered it and the latest dev version has a lot of issues with compatibility with Core (claro). 
https://git.drupalcode.org/project/gin/-/blob/fc0d2b7518b124f8ef2665a5d2ac77b782f5358b/gin.info.yml
This version did seem to work.

However the latest version has issues with library compatibility.
```
Drupal\Core\Asset\Exception\InvalidLibrariesOverrideSpecificationException
The specified library claro/classy.book-navigation does not exist
```
Which occurred here: https://git.drupalcode.org/project/gin/-/commit/4ca509d360d85c2316c8dac92fd0bb2c45880f9e


So I'd rather update to a tagged alpha version of GIN and ensure we fix all the issues that occur when doing so.

These issues occurr when going to _/admin/report/dblog_ or when trying to perform bulk actions on group members.

2. I have removed the Coffee module from the Distro. This will cause issues with the way we alter our menu's, these menu links will become available for SM through Coffee while they can't see them in the MenuTree.

If we can fix all of the above I don't mind GIN being the default admin theme. Otherwise we need to remove it as default theme and configure it per site.

**Comment by ronaldtebrake** [comment]
@agami4 Could you please still update the PR template with some nice information which will help with testing day :)

**Comment by agami4** [comment]
@ribel, I've added 'Release note' for this PR.

**Pull request merged**